### PR TITLE
Document error handling and logging

### PR DESCRIPTION
**Pull request by pcapriotti**
This PR adds a bit of developer documentation regarding how errors are propagated from a service, through the middleware, and to the client.

It also removes some unnecessary duplication of error logs and simplifies the way errors are thrown in the main `Handler → Servant.Handler` natural transformation.

## Checklist

 - [x] The **PR Title** explains the impact of the change.
 - [x] The **PR description** provides context as to why the change should occur and what the code contributes to that effect. This could also be a link to a JIRA ticket or a Github issue, if there is one.
 - [x] **changelog.d** contains the following bits of information ([details](https://github.com/wireapp/wire-server/blob/develop/docs/developer/changelog.md)):
   - [x] A file with the changelog entry in one or more suitable sub-sections. The sub-sections are marked by directories inside `changelog.d`.

### CODE REVIEW

File: services/brig/src/Brig/API/Handler.hs [hunk 1]
```diff
@@ -94,12 +93,13 @@ toServantHandler env action = do
 
     handleWaiErrors logger reqId =
       \case
-        StdError werr -> do
-          Server.logError' logger (Just reqId) werr
-          Servant.throwError $
-            Servant.ServerError (mkCode werr) (mkPhrase (WaiError.code werr)) (Aeson.encode werr) [(hContentType, renderHeader (Servant.contentType (Proxy @Servant.JSON)))]
+        -- throw in IO so that the `catchError` middleware can turn this error
```
Comment (mdimjasevic) [inline services/brig/src/Brig/API/Handler.hs:96]:
```suggestion
        -- throw in IO so that the `catchErrors` middleware can turn this error
```

**Review by mdimjasevic — APPROVED** [review]
Looks good! Just note minor remark about the name of a middleware function.

File: services/brig/src/Brig/API/Handler.hs (hunk 1 above)
Comment (pcapriotti) [reply services/brig/src/Brig/API/Handler.hs]: Fixed.

**Review by mdimjasevic — APPROVED** [review]
It's interesting to see how code can be deleted and still have the same behavior!

**Pull request merged**